### PR TITLE
Remove check for C++ compiler on Haiku

### PR DESCRIPTION
**Pull request by davidkaroly**

### CODE REVIEW

**Comment by infrastation** [comment]
Thank you trying the next step. Is there any other `CXX`-specific build logic that would be best undone in the same change?

**Comment by guyharris** [comment]
> Thank you trying the next step. Is there any other CXX-specific build logic that would be best undone in the same change?

There's stuff in configure.ac such as the early `haiku*)` case that calls `AC_PROG_CXX` and then does the "make sure the C and C++ compilers are both compiling code with the same bit-width" check.  If that's not doine in this change, it needs to be done in a separate change for autotools, but I don't see any reason to separate them.

**Comment by davidkaroly** [comment]
Yes you're right we can remove the similar stanzas from configure.ac within the same PR.

**Comment by fxlb** [comment]
> Yes you're right we can remove the similar stanzas from configure.ac within the same PR.

And in `Makefile.in` ?

**Comment by fxlb** [comment]
Like this one:
```patch
commit fcd06c0abf324388c72f09bbf54d3f971f493f39
Author: Jerome Duval <jerome.duval@gmail.com>
Date:   Tue Apr 28 18:39:03 2015 +0000

    Add Haiku pcap implementation

diff --git a/Makefile.in b/Makefile.in
--- a/Makefile.in
+++ b/Makefile.in
@@ -73,0 +73,1 @@
+CXXFLAGS = $(CCOPT) @V_LIB_CCOPT_FAT@ $(SHLIB_CCOPT) $(INCLS) $(DEFS) $(CFLAGS)
```

**Comment by fxlb** [comment]
And:
```
$ git grep -n '\<PLATFORM_CXX_SRC\>'
Makefile.in:91:PLATFORM_CXX_SRC =   @PLATFORM_CXX_SRC@
Makefile.in:101:SRC =   $(PLATFORM_C_SRC) $(PLATFORM_CXX_SRC) \
Makefile.in:106:# hack the extra indirection, and we have to handle PLATFORM_CXX_SRC
Makefile.in:108:OBJ =   $(PLATFORM_C_SRC:.c=.o) $(PLATFORM_CXX_SRC:.cpp=.o) \
configure.ac:3161:AC_SUBST(PLATFORM_CXX_SRC)
```

**Comment by guyharris** [comment]
Remove all references to `CXXFLAGS`, even in comments, from configure.ac; there should be no occurrence of the strings "CXX" or "cxx" left in configure.ac.

Remove all references to `CXXFLAGS`, `CMAKE_CXX_FLAGS`, and `CMAKE_CXX_FLAGS_MINSIZEREL` from CMakeLists.txt and cmake/Modules/FindAsan.cmake; there should be no occurrence of the strings "CXX" or "cxx" left in either of those files.

**Comment by davidkaroly** [comment]
> Remove all references to `CXXFLAGS`, even in comments, from configure.ac; there should be no occurrence of the strings "CXX" or "cxx" left in configure.ac.

Done.

> Remove all references to `CXXFLAGS`, `CMAKE_CXX_FLAGS`, and `CMAKE_CXX_FLAGS_MINSIZEREL` from CMakeLists.txt and cmake/Modules/FindAsan.cmake; there should be no occurrence of the strings "CXX" or "cxx" left in either of those files.

Done.
FindAsan seems to be embedded in CMakeLists but in any case, I removed any remaining mentions of `CXX_FLAGS` from there as well.

**Comment by infrastation** [comment]
This looks more complete, thank you. Any objections to merge?

File: CMakeLists.txt [hunk 1]
```diff
@@ -94,36 +94,6 @@ if(CMAKE_HOST_SYSTEM_NAME STREQUAL "Haiku")
     set(CMAKE_C_FLAGS_INIT "-fPIC")
 endif()
 
-#
```
Comment (guyharris) [inline CMakeLists.txt:97]: This comment should probably be left in, perhaps with "currently no C++ code" replaced with "no C++ code", to note that removing the "C" can produce weird symptoms that can take some work to diagnose.  (It also avoids doing extra C++-related work when configuring.)

Comment (guyharris) [reply CMakeLists.txt:97]: I've modified the comment in tcpdump's CMakeLists.txt, and the same change should be made to libpcap's CMakeLists.txt, so I'll just merge this change and then add the new version of the comment.

**Pull request merged**